### PR TITLE
Adapt to Base.data_pointer_from_objref removal

### PR DESCRIPTION
**Pull request by tkoolen**
Ref: https://github.com/JuliaLang/julia/pull/25362.

### CODE REVIEW

**Comment by c42f** [comment]
Looks good to me.

Looking at Base, I'm quite confused by the difference between `data_pointer_from_objectref` vs `pointer_from_objectref`, they seem to be the same all the way back to julia-0.5.

File: src/util.jl [hunk 1]
```diff
@@ -92,3 +92,9 @@ TrivialView(a::AbstractArray{T,N}) where {T,N} = TrivialView{typeof(a),T,N}(a)
 # """
 @inline drop_sdims(a::StaticArray) = TrivialView(a)
 @inline drop_sdims(a) = a
+
+@static if VERSION >= v"0.7.0-DEV.3256"
+    const _pointer_from_objref = pointer_from_objref
+else
+    const _pointer_from_objref = Base.data_pointer_from_objref
+end
```
Comment (c42f) [inline src/util.jl]: With the previous comment in mind... I assume we can just use the public `pointer_from_objectref` in julia 0.6 as well as 0.7?

Comment (tkoolen) [reply src/util.jl]: Yeah I think you're right. I thought the additional type assertion maybe had a function, but judging by https://github.com/JuliaLang/julia/commit/1750a83ce233151d29101961db9de10e74a1f624 it was just redundant.

**Comment by c42f** [comment]
Travis was having an issue with github availability.  I restarted it.

**Pull request merged**